src/Aspire.Dashboard/DashboardWebApplication.cs

```diff
@@ -148,7 +148,39 @@ public Func<EndpointInfo> OtlpServiceHttpEndPointAccessor
             // See https://learn.microsoft.com/aspnet/core/performance/response-compression#compression-with-https for more information
             options.MimeTypes = ["text/javascript", "application/javascript", "text/css", "image/svg+xml"];
         });
-        builder.Services.AddCors();
+        if (!string.IsNullOrEmpty(dashboardOptions.Otlp.Cors.AllowedOrigins))
+        {
+            builder.Services.AddCors(options =>
+            {
+                // Default policy allows the dashboard's origins.
+                // This is added so CORS middleware doesn't report failure for dashboard browser requests that include an origin header.
+                options.AddDefaultPolicy(builder =>
+                {
+                    builder.WithOrigins(dashboardOptions.Frontend.GetEndpointUris().Select(uri => uri.OriginalString).ToArray());
+                    builder.AllowAnyHeader();
+                    builder.AllowAnyMethod();
+                });
+
+                options.AddPolicy("OtlpHttp", builder =>
```

Comment:
Seems like `OtlpHttp` is a constant that's referenced again below in `OtlpHttpEndpointsBuilder.cs`. Extract to a shared location?